hax/helper/generate_sysconf.py

```diff
@@ -0,0 +1,255 @@
+        'consul', 'reload'
+    ]))
+
+
+class Service(NamedTuple):
+    id: str
+    name: str
+    address: str
+    port: int
+    checks: List[Dict]
+
+
+class Generator:
+    """
+    Generates system configuration files for consul, motr and s3 services.
+    """
+
+    def __init__(self):
+        self.kv = KVAdapter()
+        self.sysconfig_dir = '/etc/sysconfig/'
+        if not os.path.isdir(self.sysconfig_dir):
+            self.sysconfig_dir = '/etc/default/'
+
+    @repeat_if_fails()
+    def get_service_ids(self, svc_name: str) -> List[str]:
+        """
+        Returns the service id by its name assuming that it runs at the
+        current node.
+
+        Examples of the key that will match:
+            m0conf/nodes/srvnode1/processes/6/services/ha
+        """
+        node_items = self.kv.kv_get('m0conf/nodes', recurse=True)
+        node = get_node_name()
+        regex = re.compile(f'^m0conf/nodes/{node}\\/processes/([0-9]+)'
+                           f'/services/{svc_name}$')
+        keys = []
+        for key in node_items:
+            match_result = re.match(regex, key['Key'])
+            if not match_result:
+                continue
+            keys.append(match_result.group(1))
+        return keys
+
+    @repeat_if_fails()
+    def get_service_ep(self, proc_id: str) -> Optional[str]:
+        """
+        Returns the service endpoint from the given process id assuming that it
+        runs at the current node.
+
+        Examples of the key that will match:
+            key = m0conf/nodes/srvnode-1/processes/10/endpoint
+            value = 10.230.246.59@tcp:12345:2:1
+        """
+        node = get_node_name()
+        key = self.kv.kv_get(
+            f'm0conf/nodes/{node}/processes/{proc_id}/endpoint')
+        if key:
+            return key['Value'].decode("utf-8")
+        return None
+
+    def get_ios_meta_data(self, proc_id: str) -> Optional[str]:
+        """
+        Returns the metadata path for a given ioservice fid.
+
+        Examples of the key that will match:
+            key = m0conf/nodes/srvnode-1.data.private/processes/10/meta_data
+            value = /dev/vg_srvnode-1_md1/lv_raw_md1
+        """
+        node = get_node_name()
+        key = self.kv.kv_get(
+            f'm0conf/nodes/{node}/processes/{proc_id}/meta_data')
+        if key:
+            return key['Value'].decode("utf-8")
+        return None
+
+    def _write_file(self, filepath: str, contents: str) -> None:
+        logging.debug('Writing to file %s', filepath)
+        with open(filepath, 'w') as f:
+            f.write(contents)
+
+    def generate_confd(self, svc_id: str, hax_ep: str):
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        filename = f'm0d-{fid}'
+        contents = (f'MOTR_M0D_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n'
+                    'MOTR_CONF_XC=/etc/motr/confd.xc\n')
```

Comment:
Let's take motr conf dir path as an input to this function or to the Generator class itself. Avoid hard coding the paths.

Reply:
done